core/domain/story_domain.py

```diff
@@ -98,9 +100,36 @@ def __init__(self, change_dict):
         elif self.cmd == CMD_MIGRATE_SCHEMA_TO_LATEST_VERSION:
             self.from_version = change_dict['from_version']
             self.to_version = change_dict['to_version']
+        elif self.cmd == CMD_CREATE_NEW:
+            self.title = change_dict['title']
         else:
             raise Exception('Invalid change_dict: %s' % change_dict)
 
+    def to_dict(self):
+        """Returns a dict representing the StoryChange domain object.
+
+        Returns:
+            A dict, mapping all fields of StoryChange instance.
+        """
+        story_change_dict = {}
+        story_change_dict['cmd'] = self.cmd
+        if hasattr(self, 'property_name'):
```

Comment:
There's a fair amount of repetition here. Think it's worth having a class-level property

    OPTIONAL_ATTRIBUTE_NAMES = [...]

and then doing

    for attribute_name in self.OPTIONAL_ATTRIBUTE_NAMES:
        if hasattr(self, attribute_name):
            story_change_dict[attribute_name] = getattr(self, attribute_name)

perhaps?

This is optional, I'm happy with the current code too, it's a bit more explicit and sometimes more abstraction isn't the best thing anyway. So, just suggesting an alternative for your consideration, but ultimately it's your call. If you decide to go with this, make sure to do it for the other classes too.

Reply:
Yeah, this makes it look cleaner. Done.